internal/ansi_color.go

```diff
@@ -0,0 +1,35 @@
+package internal
```

Comment:
Usually, CLI tools use colors to denote something (i.e. red for failure, green for success). Since this is a progress I suggest not using colors here. It will simplify things and also make using the progress bar usable in outputs that have a tendency to be messed up by escape sequence.

Reply:
That makes perfect sense. It was good to practice using ASNI codes and adding a new file to the repo.
I'm currently working on the rest of the suggestions. Thank you for your reviewing everything -- like you said, having your code analyzed is a great opportunity.